app/src/main/java/org/mozilla/fenix/settings/PrivateBrowsingFragment.kt

```diff
@@ -41,5 +42,17 @@ class PrivateBrowsingFragment : PreferenceFragmentCompat() {
         findPreference<SwitchPreference>(getPreferenceKey(R.string.pref_key_open_links_in_a_private_tab))?.apply {
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
+
+        findPreference<Preference>(getPreferenceKey(R.string.pref_key_allow_screenshots_in_private_mode))?.apply {
+            setOnPreferenceClickListener {
+                (requireActivity() as HomeActivity).checkAndUpdateScreenshotBlockFlag()
```

Comment:
Can this cast be avoided? It adds tight coupling to an implementation we should not know about.
Could we use `SharedViewModel by activityViewModels()`?

Reply:
Actually I don't think we even need to call something in Activity.
How about using a separate object  which would know to handle enabling/disabling screenshots, one that can be used from both the activity and the fragment?